### PR TITLE
Feat: Document `HttpClient5Accessor`

### PR DESCRIPTION
**Pull request by Johannes-Schneider**
# ⚠️ DO NOT MERGE UNTIL VERSION `4.20.0` HAS BEEN PUBLISHED ⚠️

## What Has Changed?

This PR adds documentation for the new `http-client-5` module and the related features.

### CODE REVIEW

File: docs-java/features/connectivity/http-client.mdx [hunk 1]
```diff
@@ -104,13 +141,62 @@ This inheritance enables custom implementation for the following methods:
...
+        .maxConnectionsPerRoute(20)
+        .build();
+
+HttpClient genericClient = customFactory.createHttpClient();
+HttpClient destinationClient = customFactory.createHttpClient(destination);
+```
+
+The builder class (`ApacheHttpClient5FactoryBuilder`) provides the following customization options:
+
+1. `Timeout timeout` - The timeout that should be used by `HttpClient` instances created by the to-be-built `ApacheHttpClient5Factory`.
```
Comment (Johannes-Schneider) [inline docs-java/features/connectivity/http-client.mdx:160]:
```suggestion
1. `Duration timeout` - The timeout that should be used by `HttpClient` instances created by the to-be-built `ApacheHttpClient5Factory`.
```

File: docs-java_versioned_docs/version-v5/features/connectivity/http-client.mdx [hunk 2]
```diff
@@ -104,13 +141,62 @@ This inheritance enables custom implementation for the following methods:
...
+        .maxConnectionsPerRoute(20)
+        .build();
+
+HttpClient genericClient = customFactory.createHttpClient();
+HttpClient destinationClient = customFactory.createHttpClient(destination);
+```
+
+The builder class (`ApacheHttpClient5FactoryBuilder`) provides the following customization options:
+
+1. `Timeout timeout` - The timeout that should be used by `HttpClient` instances created by the to-be-built `ApacheHttpClient5Factory`.
```
Comment (Johannes-Schneider) [inline docs-java_versioned_docs/version-v5/features/connectivity/http-client.mdx:160]:
```suggestion
1. `Duration timeout` - The timeout that should be used by `HttpClient` instances created by the to-be-built `ApacheHttpClient5Factory`.
```

File: docs-java/faq.mdx [hunk 3]
```diff
@@ -48,7 +48,7 @@ Due to an [incompatible change in the Spring Framework 6](https://github.com/spr
 The issue is caused by a change to the `RestTemplate`, which now requires Apache Http Client version 5 (previously required version 4).
 The new http client version is not downwards compatible and the Cloud SDK currently still relies on `RestTemplate` with Http Client version 4.
 
-This also affects use cases where `RestTemplate` is used directly with the [Http Client provided by the SAP Cloud SDK](/docs/java/features/connectivity/http-client).
+This **does not** affect use cases where `RestTemplate` is used directly with the [Http Client provided by the SAP Cloud SDK](/docs/java/features/connectivity/http-client).
```
Comment (newtork) [inline docs-java/faq.mdx:51]: Not yet in production.

Comment (Johannes-Schneider) [reply docs-java/faq.mdx:51]: No, but it will be with the next release (i.e. version `4.20.0`)

**Review by MatKuhr — APPROVED** [review]

**Pull request merged**